### ISSUE TITLE
Docker first setup error

### ISSUE THREAD

**Issue by antoine2tt** [eikek/sharry#482]
Hello ! 

When i'm first trying to run the container, i have the following issue : 

> sharry        | 16:28:57,270 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
> sharry        | 16:28:57,270 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
> sharry        | 16:28:57,271 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/opt/sharry-restserver-1.8.0/lib/com.github.eikek.sharry-restserver-1.8.0.jar!/logback.xml]
> sharry        | 16:28:57,280 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@5ef60048 - URL [jar:file:/opt/sharry-restserver-1.8.0/lib/com.github.eikek.sharry-restserver-1.8.0.jar!/logback.xml] is not of type file
> sharry        | 16:28:57,391 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
> sharry        | 16:28:57,393 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
> sharry        | 16:28:57,398 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
> sharry        | 16:28:57,413 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sharry] to DEBUG
> sharry        | 16:28:57,413 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [emil] to DEBUG
> sharry        | 16:28:57,413 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.http4s] to INFO
> sharry        | 16:28:57,413 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to WARN
> sharry        | 16:28:57,413 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
> sharry        | 16:28:57,414 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
> sharry        | 16:28:57,415 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@30ee2816 - Registering current configuration as safe fallback point
> sharry        | 2021-06-15 16:28:57,597 [main] INFO  s.r.Main - Using given config file: /opt/sharry.conf 
> sharry        | Exception in thread "main" pureconfig.error.ConfigReaderException: Cannot convert configuration to a sharry.restserver.Config. Failures are:
> sharry        |   - (/opt/sharry.conf) /opt/sharry.conf: java.io.FileNotFoundException: /opt/sharry.conf (Is a directory).
> sharry        | 
> sharry        |         at pureconfig.ConfigSource.loadOrThrow(ConfigSource.scala:73)
> sharry        |         at pureconfig.ConfigSource.loadOrThrow$(ConfigSource.scala:70)
> sharry        |         at pureconfig.ConfigSource$$anon$2.loadOrThrow(ConfigSource.scala:303)
> sharry        |         at sharry.restserver.ConfigFile$.loadConfig(ConfigFile.scala:19)
> sharry        |         at sharry.restserver.Main$.run(Main.scala:44)
> sharry        |         at cats.effect.IOApp.$anonfun$main$3(IOApp.scala:69)
> sharry        |         at cats.effect.internals.IOAppPlatform$.mainFiber(IOAppPlatform.scala:45)
> sharry        |         at cats.effect.internals.IOAppPlatform$.main(IOAppPlatform.scala:27)
> sharry        |         at cats.effect.IOApp.main(IOApp.scala:69)
> sharry        |         at cats.effect.IOApp.main$(IOApp.scala:68)
> sharry        |         at sharry.restserver.Main$.main(Main.scala:13)
> sharry        |         at shar

I'm using the default docker compose file : https://github.com/eikek/sharry/blob/master/docker/docker-compose.yml

And i only did : docker-compose up -d 

Did i miss something ?

Help is appreciated ! 

Thank you !

**Comment by eikek** [comment]
Hi @antoine2tt ,
hm I'm not sure what's wrong…. The config files seems to be mounted as a directory and not as a file, which is strange. I think this might happen if the file doesn't exist. Which directory were you in when you executed the `docker-compose up -d` command and does a file `sharry.conf` exists? I just tried it here with the current master branch and it worked for me. I changed into the `docker/` directory first before running `docker-compose up`.

**Comment by antoine2tt** [comment]
Hi @eikek 

Thanks for your answer 

I've made a few fails, now it's ok, the container is starting. 

I just have another issue (upload stuck at 0%), should i make another issue on github ?

**Comment by eikek** [comment]
We can sort this out here. Does this happen right from the start? So does this happen to all files - you can't upload any file to more than 0%?

**Comment by eikek** [comment]
Hi @antoine2tt do you still have problems or can this be closed maybe?

**Comment by antoine2tt** [comment]
Hey ! 

Yes you can close it

**Comment by fboterblom** [comment]
Hey antoine2tt,

I have the same problem, how did you solved it?

**Comment by antoine2tt** [comment]
> Hey antoine2tt,
> 
> 
> 
> I have the same problem, how did you solved it?
> 
> 
> 
> 

Well, the truth is that I never solved it. I moved to another project ^^

**Comment by fboterblom** [comment]
ah, thats also a solution :)

**Comment by fboterblom** [comment]
@antoine2tt which project you moved to?

**Comment by joost00719** [comment]
![image](https://user-images.githubusercontent.com/21021559/176501641-8e29c059-2b1f-4b67-95d9-25c8f7ec7316.png)
Same issue for me.

**Comment by eikek** [comment]
Hello @joost00719 this error says that `/opt/sharry.conf` is a directory and not a file. This happens for example if the source file that is being mounted does not exist. The startup works fine for me when I go into the `docker` directory and run `docker compos up -d`. I guess therefore something is different on your setup. From your screenshot it looks like this is windows? maybe you need to adopt the path in the dockerfile for your OS

**Comment by joost00719** [comment]
Hey Eikek, I got it to work 15 minutes ago on a linux server. However after starting the container it would never listen to any http requests (using nginx as reverse proxy).
I did not try to expose the ports directly and access it directly, but I can do that tomorrow.

**Comment by otherjoel** [comment]
> The startup works fine for me when I go into the `docker` directory and run `docker compose up -d`.

I was having the same error, and was not until I found this issue that I tried adding the `-d` flag, which fixed it. I suggest the docs be updated.

**Comment by eikek** [comment]
The `-d` only puts the process(es) in the background if I remember correctly. docker-compose requires a directory to be used as the "project directory" (there are docs about docker-compose online). If it is not give via an option, it uses the current working directory. Maybe on your OS it requires to explicitely set it via `PWD`.

**Comment by otherjoel** [comment]
This is Ubuntu 20.04 LTS. Issue is moot for me at this point since I had much more success just running sharry directly, without Docker.

**Comment by Fox51** [comment]
One year later, I could run the project on portainer, by first creating the file sharry.conf and then creating the stack with the docker compose  YML file, of course, modifying it with my needs. It would be great to add a hatcheck for the container, like this:

--For db--
healthcheck:
    test: [ "CMD-SHELL", "pg_isready -d $DB_NAME -U $DB_USER" ]
    interval: 30s
    timeout: 60s
    retries: 5
    start_period: 60s 

--For backend--
healthcheck:
    test: wget --no-verbose --tries=1 --spider http://localhost:9090/api/v2/open/info/version || exit 1
    interval: 60s
    timeout: 60s
    retries: 2
    start_period: 30s
